src/app/shapes/Ellipse.ts

```diff
@@ -0,0 +1,35 @@
+import * as Konva from "konva";
+import {Shape} from "./Shape";
+
+export class Ellipse extends Shape{
+
+  public instance: Konva.Ellipse;
+  
+  public constructor(name: string, width: number, height: number) {
+        super(name);
+        this.instance = this.createKonvaEllipse(width, height);
+  }
+
+  public getLeft(): number {
+    return this.instance.x() - this.instance.radius().x();
+  }
+  
+  public getTop(): number {
+    return this.instance.y() - this.instance.radius().y();
+  }
+  
+  private createKonvaEllipse(width: number, height: number): Konva.Ellipse {
+      const strokeWidth = 2;
+    return new Konva.Ellipse({
+        x: (width - (2*strokeWidth) )/ 2 + strokeWidth,
```

Comment:
What is happening here? Can we make a private method with a descriptive name to say what is happening here? Also why are we adding the strokeWidth at the end?